tests/checkpointing/unit/test_async_writer.py

```diff
@@ -219,3 +228,80 @@ def test_cached_metadata(self, tmp_path_dist_ckpt, async_queue):
                 ), f'{field.name} is different in metadata from non-cached, cached metadata impls'
         ckpt_dir.cleanup()
         async_queue.close()
+
+    def test_async_cp_with_multiple_queue_and_abort(self, tmp_path_dist_ckpt):
+        """
+        Verifies that async checkpointing backend can be used with multiple async queues.
+        For example, user may want to save 2 checkpoints i.e. one sharded state and one only on rank-0.
+        Verify the abort CP functionality and the ability to resume after an abort operation
+        """
+        Utils.initialize_distributed()
+        model = FSDP(Model((1024, 1024), 8))
+        async_queue_dist = AsyncCallsQueue()
+        ckpt_impl = TorchAsyncCheckpoint(persistent_queue=True)
+        with (
+            TempNamedDir(
+                tmp_path_dist_ckpt / 'async_checkpoint_dist', sync=True
+            ) as async_ckpt_dir_dist,
+            TempNamedDir(
+                tmp_path_dist_ckpt / 'async_checkpoint_no_dist', sync=True
+            ) as async_ckpt_dir_no_dist,
+        ):
+            state_dict = model.state_dict()
+            planner = DefaultSavePlanner()
+
+            # Perform async saves for both dist CP and non-dict CP use cases.
+            self.async_save_checkpoint(async_ckpt_dir_dist, state_dict, planner, async_queue_dist)
+            self.async_save_checkpoint_on_rank0(async_ckpt_dir_no_dist, state_dict, ckpt_impl)
+            async_queue_dist.maybe_finalize_async_calls(blocking=True, no_dist=False)
+            ckpt_impl.finalize_async_save(blocking=True, no_dist=True)
+
+            # Abort the CP workers to mock the action of inprocess restarts
+            abort_nvrx_checkpoint()
+
+            # validate state of the Async CP workers after abort operation
+            async_calls_queue_no_dist = ckpt_impl._get_async_calls_queue()
+            assert (
+                async_calls_queue_no_dist is not None
+            ), "We expect a valid state of AsyncCallsQueue"
+            async_process_no_dist = async_calls_queue_no_dist._get_async_caller()
+            if async_process_no_dist is not None:
+                assert (
+                    async_process_no_dist._debug_is_async_process_running() is False
+                ), "After abort async process must stop"
+
+            assert async_queue_dist is not None, "We expect a valid state of AsyncCallsQueue"
```

Comment:
I am confused by these assertions. How could `async_queue_dist` possibly be `None` here? How does asserting that it is not `None` verify that it is in a valid state?